vendor/src/github.com/docker/engine-api/types/container/config.go

```diff
@@ -24,6 +38,7 @@ type Config struct {
 	StdinOnce       bool                  // If true, close stdin after the 1 attached client disconnects.
 	Env             []string              // List of environment variable to set in the container
 	Cmd             strslice.StrSlice     // Command to run when starting the container
+	Healthcheck     *HealthConfig         `json:",omitempty"` // Command to run to check container is healthy
```

Comment:
I see complexity this can bring on. I think the issue is that you may want to define a tcp, http and cmd based healthcheck. Perhaps, it is sufficient to let these coexist in a single command, eliding the need to have complex merge semantics of health declarations (and having to name the health checks to tell them apart).